pkg/gatewayserver/io/io.go

```diff
@@ -265,7 +269,22 @@ func (c *Connection) ScheduleDown(path *ttnpb.DownlinkPath, msg *ttnpb.DownlinkM
 	if err != nil {
 		return 0, err
 	}
-	phy, err := band.GetByID(c.fp.BandID)
+	var fp *frequencyplans.FrequencyPlan
+	for _, gtwFP := range c.fps {
+		reqFP, err := c.fpStore.GetByID(request.FrequencyPlanID)
+		if err != nil {
+			return 0, err
+		}
+		if gtwFP.BandID == reqFP.BandID {
+			fp = gtwFP
+			break
+		}
+	}
+	if fp == nil {
+		return 0, errFrequencyPlanNotConfigured
```

Comment:
Missing attribute

Reply:
Also, we don't break API here; if there's no frequency plan in the `TxRequest`, we do what we did before

Please keep (some) test cases with the current behavior to test that

Reply:
That's a good one. Updated.